File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumOfLeftLeaves(TreeNode root) {
        return helper(root, false);    
    }
    
    public int helper(TreeNode node, boolean isLeft) {
        if (node.left == null && node.right == null) {
            // leaf
            return isLeft? node.val : 0;
        }
        if (node.left == null) {
            return helper(node.right, false);
        }
        if (node.right == null) {
            return helper(node.left, true);
        }
        return helper(node.left, true) + helper(node.right, false); 
    }
    
    
}

File: 0404-sum-of-left-leaves/0404-sum-of-left-leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        sum = 0
        level = [root]
        next_level = []
        
        while level:
            for node in level:
                left = node.left
                if left:
                    if not left.left and not left.right: # leaf
                        sum += left.val
                    else:
                        next_level.append(left)
                right = node.right
                if right:
                    next_level.append(right)
            level = next_level
            next_level = []
        
        return sum
                    
                    
                            
                    
        